src/Microsoft.ServiceFabric.Actors.KVSToRCMigration/Extensions/KvsActorStateProviderExtensions.cs

```diff
@@ -211,6 +195,54 @@ internal static bool GetDisableTombstoneCleanupSetting(this KvsActorStateProvide
             return stateProvider.GetStoreReplica().KeyValueStoreReplicaSettings.DisableTombstoneCleanup;
         }
 
+        internal static Task GetValueByKeysAsync(this KvsActorStateProvider stateProvider, List<string> keys, HttpResponse response, CancellationToken cancellationToken)
+        {
+            var pairs = new List<KeyValuePair>();
+            return stateProvider.GetActorStateProviderHelper().ExecuteWithRetriesAsync(
+                        async () =>
+                        {
+                            if (keys.Any())
+                            {
+                                using var tx = stateProvider.GetStoreReplica().CreateTransaction();
+
+                                foreach (var key in keys)
+                                {
+                                    var result = stateProvider.GetStoreReplica().TryGet(tx, key);
+
+                                    if (result == null)
+                                    {
+                                        throw new ActorStateMigratedDataValidationFailedException($"Could not find key: {key} in KVS");
+                                    }
+
+                                    pairs.Add(new KeyValuePair() { Key = key, Value = result.Value });
+                                }
+                            }
+
+                            await WriteKeyValuePairsToResponse(pairs, response);
+                        },
+                        "GetValueByKeyAsync",
+                        cancellationToken);
+        }
+
+        private static async Task WriteKeyValuePairsToResponse(List<KeyValuePair> pairs, HttpResponse response)
+        {
+            using var memoryStream = new MemoryStream();
+            var binaryWriter = XmlDictionaryWriter.CreateTextWriter(memoryStream);
+            keyValuePairSerializer.WriteObject(binaryWriter, pairs);
+            binaryWriter.Flush();
+
+            var byteArray = memoryStream.ToArray();
+            var newLine = Encoding.ASCII.GetBytes("\n");
+
+            ActorTrace.Source.WriteNoise("KvsActorStateProviderExtensionHelper", $"ByteArray: {byteArray} ArrayLength: {byteArray.Length} StreamLength: {memoryStream.Length}");
+
+            // Set the content type
+            response.ContentType = "application/xml; charset=utf-8";
```

Comment:
ContentType should be set only if doesnt exist. Otherwise this would cause failures if the response is spanned across multiple chunks